lib/XMLParser.php

```diff
@@ -72,14 +72,16 @@ public function parse($xml)
         while ($this->read()) {
             switch ($this->nodeType) {
                 case \XMLReader::ELEMENT:
+                    // Take a copy of this before it is lost
+                    $isEmpty = $this->isEmptyElement;
                     $this->path[] = $this->name;
                     if ($this->startElementCallback) {
                         call_user_func($this->startElementCallback, $this);
                     }
-                    if ($this->isEmptyElement) {
-                        array_pop($this->path);
+                    if (!$isEmpty) {
+                        break;
                     }
-                    break;
+                    // ELSE: if element is empty, also call End Element callback
```

Comment:
This line is not quite clear to me. What do you mean with "End Element callback" here? You mean that it will takes `case \XMLReader::END_ELEMENT:` on the next turn?

Reply:
I added the comment because it is slightly unclear what is happening - it is kind of a weird behaviour from C.

If you come to an empty XML element (ie `<foaf:homepage rdf:resource="http://www.example.com/joe/" />`). Then it will run the code both in the `case \XMLReader::START_ELEMENT` block and because it never calls `break`, it will also run the code in the`case \XMLReader::END_ELEMENT` block.

It might be unexpected to a developer looking at the code but it is quite an efficient way of doing it...

Reply:
Thank you for clarifying.